src/editors/containers/ProblemEditor/data/OLXParser.js

```diff
@@ -634,14 +670,16 @@ export class OLXParser {
       throw new Error('Misc Attributes asscoiated with problem, opening in advanced editor');
     }
 
+    const problemType = this.getProblemType();
+
+    this.checkTextAfterProblemTypeTag(problemType);
```

Comment:
When reading the code, it's not immediately obvious to me that this function can throw an error that errors out of the running code. What do you think about making it explicit, like in the line above with `throw new Error('Misc Attributes asscoiated with problem, opening in advanced editor');`?

For example, since the error message here is the same whether you have invalid keys or invalid tags, the `checkTextAfterProblemTypeTag` can just be something like `hasXMLAfterProblemTypeTag` and then here you could do
```
if (this.hasXMLAfterProblemTypeTag(problemType) {
  throw new Error(`OLX found after the ${problemType} tags, opening in advanced editor`);
}
```
. The benefit of doing it in a similar way to what I suggested is that any reader of the code can immediately see that an error is thrown, which makes the code clearer and the logic less confusing in my opinion.